### PR TITLE
virts-163

### PR DESCRIPTION
**Pull request by ArtificialErmine**
Implements operation pausing and canceling for chain style operations.

### CODE REVIEW

File: templates/chain.html [hunk 1]
```diff
@@ -234,34 +234,74 @@ <h4>Manage operations</h4>
...
+                                                               </button>
+                                                                <button id="control-pause" class="control-row-button" onclick="controlOp('pause')">
+                                                                   <span class="icon"><img src="/chain/img/pause.png"></span>
+                                                                   <span class="caption">Pause Operation</span>
+                                                               </button>
+                                                            </td>
+                                                            <td>
+                                                                <button class="control-row-button" onclick="controlOp('cancel')">
+                                                                   <span class="icon"><img src="/chain/img/cancel.png"></span>
+                                                                   <span class="caption">Cancel Operation</span>
```
Comment (khyberspache) [inline templates/chain.html]: I think we might want to tweak the Execution Control UI element... it seems a little out of place.  Maybe make the entire table part of the flex grid (inline-flex), set a min-width, and put the buttons above the PAUSED/RESUME/CANCELLED display box.

File: static/js/chain.js [hunk 2]
```diff
@@ -25,3 +25,35 @@ function showHide(show, hide) {
...
+            success: function (data) {
+                if (data['result'] === 'PAUSED'){
+                    $('#control-play').css('display','');
+                    $('#control-pause').css('display','none');
+                }
+                if (data['result'] === 'RUNNING') {
+                    $('#control-play').css('display','none');
+                    $('#control-pause').css('display','');
+                }
+                document.getElementById('control-state').innerHTML = data['result']
```
Comment (khyberspache) [inline static/js/chain.js]: These AJAX requests should adhere to the front-to-back-end call structure we've agreed to: https://confluence.mitre.org/display/CAL/CALDERA+-+Chain+Mode%3A++Front-end+to+Back-end+API+Calls

Comment (khyberspache) [reply static/js/chain.js]: Let's update the JS function to include a Default endpoint & override ability like:
`function restRequest(type, data, callback, endpoint='/plugin/chain/rest'){//code}`

File: static/js/sections.js [hunk 3]
```diff
@@ -149,6 +152,7 @@ function refresh() {
     let selectedOperationId = $('#operations option:selected').attr('value');
     let postData = selectedOperationId ? {'index':'core_operation','id': selectedOperationId} : null;
     restRequest('POST', postData, operationCallback);
+    getOpState();
```
Comment (khyberspache) [inline static/js/sections.js]: using the updated restRequest, do something like:
```
let op = $('#operations option:selected').attr('value');
restRequest('POST', {'id': op, 'mode': 'state'}, opControlCallback, '/op/control') 
```

**Comment by privateducky** [comment]
per prior caldera comment; closing virts-163 PRs so we can re-assess the functionality we're looking to add.

**Pull request closed without merging**